runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -1190,12 +1190,12 @@ public Timer withOutputTimestamp(Instant outputTimestamp) {
      * </ul>
      */
     private void setAndVerifyOutputTimestamp() {
-
       if (outputTimestamp != null) {
         checkArgument(
-            !outputTimestamp.isBefore(elementInputTimestamp),
```

Comment:
Do we need to make a similar change in other locations like `FnApiDoFnRunner.java`?

Reply:
So I ran the new validates runner tests against FnApiDoFnRunner. Turns out it was not properly accounting for DoFN skew when setting timers. But it wasn't doing any checking on timestamps at all for normal DoFn emit.

I added checks for the normal DoFn case and allowed for skew in the timer case.